src/analyzer/protocol/ssl/spicy/SSL.spicy

```diff
@@ -0,0 +1,1901 @@
+  key_share = 51, # new for 1.3, see RFC 8446
+  transparency_info = 52, # temporary - draft-ietf-trans-rfc6962-bis-34
+  connection_id = 53, # temporary -d draft-ietf-tls-dtls-connection-id
+  external_id_hash = 55, # temporary - RFC-ietf-mmusic-sdp-uks-07
+  external_session_id = 56, # temporary - RFC-ietf-mmusic-sdp-uks-07
+  quic_transport_parameters = 57, # temporary - draft-ietf-quic-tls-32
+  next_protocol_negotiation = 13172,
+  origin_bound_certificates = 13175,
+  encrypted_client_certificates = 13180,
+  channel_id = 30031,
+  channel_id_new = 30032,
+  padding_old = 35655,
+  renegotiation_info = 65281
+};
+
+type ECPointFormat = enum {
+ uncompressed = 0,
+ ansiX962_compressed_prime = 1,
+ ansiX962_compressed_char2 =2
+};
+
+type ECCurveType = enum {
+  EXPLICIT_PRIME = 1,
+  EXPLICIT_CHAR = 2,
+  NAMED_CURVE = 3
+};
+
+type NamedCurve = enum {
+  sect163k1 = 1, sect163r1 = 2, sect163r2 = 3,
+  sect193r1 = 4, sect193r2 = 5, sect233k1 = 6,
+  sect233r1 = 7, sect239k1 = 8, sect283k1 = 9,
+  sect283r1 = 10, sect409k1 = 11, sect409r1 = 12,
+  sect571k1 = 13, sect571r1 = 14, secp160k1 = 15,
+  secp160r1 = 16, secp160r2 = 17, secp192k1 = 18,
+  secp192r1 = 19, secp224k1 = 20, secp224r1 = 21,
+  secp256k1 = 22, secp256r1 = 23, secp384r1 = 24,
+  secp521r1 = 25,
+  arbitrary_explicit_prime_curves = 0xFF01,
+  arbitrary_explicit_char2_curves = 0xFF02
+};
+
+type HeartbeatMode = enum {
+ peer_allowed_to_send = 1,
+ peer_not_allowed_to_send = 2
+};
+
+type NameType = enum {
+  host_name = 0
+};
+
+type AlertLevel = enum {
+  warning= 1, fatal= 2
+};
+
+type AlertDescription = enum {
+  close_notify = 0,
+  unexpected_message = 10,
+  bad_record_mac = 20,
+  decryption_failed_RESERVED = 21,
+  record_overflow = 22,
+  decompression_failure = 30,
+  handshake_failure = 40,
+  no_certificate_RESERVED = 41,
+  bad_certificate = 42,
+  unsupported_certificate = 43,
+  certificate_revoked = 44,
+  certificate_expired = 45,
+  certificate_unknown = 46,
+  illegal_parameter = 47,
+  unknown_ca = 48,
+  access_denied = 49,
+  decode_error = 50,
+  decrypt_error = 51,
+  export_restriction_RESERVED = 60,
+  protocol_version = 70,
+  insufficient_security = 71,
+  internal_error = 80,
+  user_canceled = 90,
+  no_renegotiation = 100,
+  unsupported_extension = 110
+};
+
+type HashAlgorithm = enum {
+  none = 0, md5= 1, sha1= 2, sha224= 3, sha256= 4, sha384= 5,
+  sha512 = 6
+};
+
+type SignatureAlgorithm = enum {
+  anonymous = 0, rsa= 1, dsa= 2, ecdsa= 3
+};
+
+type TLSVersion = enum {
+  NONE = 0xFFFFFF, # out of range, only used internally
+  SSLv2  = 0x0002,
+  SSLv3  = 0x0300,
+  TLSv10 = 0x0301,
+  TLSv11 = 0x0302,
+  TLSv12 = 0x0303,
+  DTLSv10   = 0xFEFF,
+  # DTLSv11 does not exist.
+  DTLSv12   = 0xFEFD,
+  DTLSv13   = 0xFEFC
+};
+
+const UNKNOWN_VERSION : uint16 = 0x0000;
+const SSLv2 = 0x0002;
+const SSLv3  = 0x0300;
+const TLSv10 = 0x0301;
+const TLSv11 = 0x0302;
+const TLSv12 = 0x0303;
+const TLSv13 = 0x0304;
```

Comment:
The constants here seem to duplicate some of the values in `TLSVersion` above, but e.g., `TLSv13` seems to be missing. Could you get rid of the constants and exclusively use the enum values instead?

Reply:
You are right that this is duplicate. Due to the way that these are used in the analyzer, it is easier to get rid of the enum, and instead just use the consts everywhere.

It is possible to use the enums instead - but this would necessitate casts to their int-value in essentially all locations. Which at the very least looks ugly/bulky.